consensus/src/aggregation/mod.rs

```diff
@@ -0,0 +1,1222 @@
+//! Recover threshold signatures to represent agreement over an ordered sequence of items.
+//!
+//! This module allows a dynamic set of participants to collectively produce threshold signatures
+//! for any ordered sequence of items.
+//!
+//! A primary use case for this is to allow blockchain validators to agree on a series of digests,
+//! such as state roots. Some chains may finalize transaction data but not execute the transactions
+//! during consensus. This module enables validators to come to cryptographically-proven agreement
+//! on the state root, which is necessary to support features like state sync.
```

Comment:
nit: link ordered-broadcast if you want participants to be able to drive their own sequences.